test_shop.py

```diff
@@ -1,4 +1,5 @@
-# Unit testing
+# Unit testing 
+#Final changes for pull request
```

Comment:
with self.assertRaises(MaximumPurchaseAttemptsExceededError): - error not raised so test fails